### PR TITLE
Future planning doc

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Collection notes from planning meeting and email conversations.

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
First draft, cleaning up formatting and language.

File: doc/planning.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# gRPC 5.0 planning
+
+| Legend |                          |
+|--------|--------------------------|
+| 🔴      | Stretch goal |
```
Comment (jtattermusch) [inline doc/planning.md]: nit: are these symbols some standard you're using for ASP.NET core planning? They seem a bit unnecessary to me, just `[stretch]` prefix would probably suffice. Leaving up to you.

File: doc/planning.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+# gRPC 5.0 planning
```
Comment (jtattermusch) [inline doc/planning.md]: Rephrase as 
"grpc-dotnet future plans" or "grpc-dotnet for ASP.NET core 5" or sth in that sense.
- mention it's plans for ASP.NET core 5 and perhaps a bit of additional context.
- add the approximate ETA

It likely won't be gRPC "5.0" (and if so, using such version number is subject to approval)

File: doc/planning.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+ - Ensure code-first gRPC is successful
+  - Marc Gravell is using protobuf.net to build a code-first gRPC experience on grpc-dotnet. Ensure our abstractions allow him to be successful.
+
+ - Ensure WCF to gRPC ebook is successful
+  - Provide SME guidance where applicable
+
+- Testing existing client libraries on both implementations
+    - There are APIs and SDKs to share between the implementations
+    - @Jan Tattermusch you mentioned that you have a doc noting the differences that you can share?
```
Comment (jtattermusch) [inline doc/planning.md]: https://github.com/grpc/grpc-dotnet/blob/master/doc/implementation_comparison.md

**Comment by JunTaoLuo** [comment]
Oops, I realized I messed up my setup and created a branch in the main repo instead of my own fork. I'm going to redo the PR with my own fork and will address the comments there.

**Pull request closed without merging**